src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs

```diff
@@ -139,11 +140,13 @@ public bool Exists(string indexName)
 
         public async Task StoreDocumentsAsync(string indexName, IEnumerable<DocumentIndex> indexDocuments)
         {
+            var luceneIndexSettings = await _luceneIndexSettingsService.GetSettingsAsync(indexName);
+
             await WriteAsync(indexName, writer =>
             {
                 foreach (var indexDocument in indexDocuments)
                 {
-                    writer.AddDocument(CreateLuceneDocument(indexDocument));
+                    writer.AddDocument(CreateLuceneDocument(indexDocument, luceneIndexSettings));
```

Comment:
@Piedone Here, what I ended up doing is moving the awaited method outside of CreateLuceneDocument. This way we don't need to await the writer's delegate.

It was not making any sense to call the _luceneIndexSettingsService on every item in the `foreach`.